dts/arm/atmel/saml2x-pinctrl.dtsi

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2021 Argentum Systems Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "pinctrl_atmel_sam.h"
+
+/ {
+	soc {
+		pinctrl@40002800 {
+			/* instance, signal, port, pin, peripheral [, flags] */
+			DT_ATMEL_PORT(sercom0, pad0, a,  4, d, pinmux-enable);
+			DT_ATMEL_PORT(sercom0, pad1, a,  5, d, pinmux-enable);
+			DT_ATMEL_PORT(sercom0, pad2, a,  6, d, pinmux-enable);
+			DT_ATMEL_PORT(sercom0, pad3, a,  7, d, pinmux-enable);
+			DT_ATMEL_PORT(sercom0, pad0, a,  8, c, pinmux-enable);
+			DT_ATMEL_PORT(sercom0, pad1, a,  9, c, pinmux-enable);
+			DT_ATMEL_PORT(sercom0, pad2, a, 10, c, pinmux-enable);
+			DT_ATMEL_PORT(sercom0, pad3, a, 11, c, pinmux-enable);
```

Comment:
gentle reminder: all platforms should start migrating to pinctrl, ref https://github.com/zephyrproject-rtos/zephyr/issues/39740. The format used here (macros) should not be used.

Reply:
@gmarull , this platform was there before this rule be created and, as far I remember, this could stay as is.

Reply:
The Atmel pinmux driver came before pinctrl, so this topic was not discussed at that time. During pinctrl implementation it was concluded that such "helpers" were not a good idea because:

1. They hide Devicetree language
2. Could make external tooling (e.g. editor extensions) more difficult to implement

So, I'd prefer not to have yet another flavor of Devicetree representation unless it is strictly needed.

Reply:
@gmarull , As far I remember, it was decided that vendor is free to choose about devicetree representation related to pinctrl. This is not a new flavor, it was the first pinctrl version to be in the tree.

Zephyr still on migrate phase and I think it is not appropriated to block something that is already in the tree.

On my view, I think most important is focus to drop pinmux drivers ASAP and after that work on a migration path.

Reply:
So this case is not about changing DT operation mode really (boards will not require any update). It's about not using macros that expand to DT nodes.

Btw, the current format can be easily converted using a Python script like that:

```python
import re
import sys

output = ""
with open(sys.argv[1], "r") as f:
    for line in f:
        m = re.match(
            r"(\s+)DT_ATMEL_PORT\((.*),\s+(.*),\s+(.*),\s+(.*),\s+(.*),\s+(.*)\);", line
        )
        if not m:
            output += line
            continue

        args = "\n".join(
            (f"{m.group(1)}\t{arg.strip()};" for arg in m.group(7).split(","))
        )

        output += f"{m.group(1)}p{m.group(4)}{m.group(5)}{m.group(6)}_{m.group(2)}_{m.group(3)} {{\n"
        output += f"{m.group(1)}\tatmel,pins = <&port{m.group(4)} {m.group(5)} PERIPH_{m.group(6)}>;\n"
        output += args
        output += f"\n{m.group(1)}}};\n\n"

with open(sys.argv[1], "w") as f:
    f.write(output)
```

which converts content like:
```
			DT_ATMEL_PORT(sercom0, pad0, a,  4, d, pinmux-enable);
			DT_ATMEL_PORT(sercom0, pad1, a,  5, d, pinmux-enable);
```

to:

```
			pa4d_sercom0_pad0 {
				atmel,pins = <&porta 4 PERIPH_d>;
				pinmux-enable;
			};

			pa5d_sercom0_pad1 {
				atmel,pins = <&porta 5 PERIPH_d>;
				pinmux-enable;
			};
```

so the effort to align with pinctrl is really low.

Reply:
No, what you suggests is not a final solution and any new pin to be added will imply in a lot of more effort. Zephyr and Devicetree itself are constructed above macro expansion. Here uses same principle, so I don't see any motive to not use it. There are pros/cons on each approach and the macro still better for Atmel use cases.

I prefer keep macro to define a pin until have a final solution, which will be similar to how STM/GD have been defining pins.

Reply:
Are we agreeing to continue using `DT_ATMEL_PORT()`? If no, then #42512 (and a lot of other SAM0 sources) will require some rework too...